### PR TITLE
ARM split - Add support for dfu-util EE_HANDS flashing

### PR DESCRIPTION
**Pull request by zvecr**
## Current Issue
No method to set eeprom handedness

## Description
Existing AVR implementations rely on flashing another file to set a particular eeprom byte.

ARM currently looks like it has no way to configure EECONFIG_HANDEDNESS within the emulated eeprom via dfu-util or another tool.

Until a solution is found, this PR emulates the behaviour with the limitation that produces different binaries for the left and right sides.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
## Testing notes
Scenarios currently tested
* f303

File: tmk_core/common/eeconfig.c [hunk 1]
```diff
@@ -50,6 +50,13 @@ void eeconfig_init_quantum(void) {
   eeprom_update_dword(EECONFIG_RGB_MATRIX,    0);
   eeprom_update_byte(EECONFIG_RGB_MATRIX_SPEED, 0);
 
+  // TODO: Remove once ARM has a way to configure EECONFIG_HANDEDNESS
```
Comment (zvecr) [inline tmk_core/common/eeconfig.c]: An alternative solution is to remove the logic from eeprom init, and relocate it to `is_keyboard_left` within `split_util.c`.

File: docs/newbs_flashing.md [hunk 2]
```diff
@@ -298,6 +298,16 @@ File downloaded successfully
...
 ```
 
+#### STM32 Commands
+
+There are a number of DFU commands that you can use to flash firmware to a STM32 device:
+
+* `:dfu-util` - The default command for flashing to STM32 devices. 
+* `:dfu-util-wait` - This works like the default command, but it gives you a (configurable) 10 second timeout before it attempts to flash the firmware.  You can use `TIME_DELAY=20` from the command line to change the timeout.
+   * Eg: `make <keyboard>:<keymap>:dfu-util TIME_DELAY=5`
+* `:dfu-utilsplit-left` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Left Side" EEPROM setting for split keyboards.
```
Comment (fauxpark) [inline docs/newbs_flashing.md]:
```suggestion
* `:dfu-util-split-left` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Left Side" EEPROM setting for split keyboards.
```

File: docs/newbs_flashing.md [hunk 3]
```diff
@@ -298,6 +298,16 @@ File downloaded successfully
...
+* `:dfu-utilsplit-right` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Right Side" EEPROM setting for split keyboards.
```
Comment (fauxpark) [inline docs/newbs_flashing.md]:
```suggestion
* `:dfu-util-split-right` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Right Side" EEPROM setting for split keyboards.
```

File: docs/flashing.md [hunk 4]
```diff
@@ -174,4 +174,6 @@ There are a number of DFU commands that you can use to flash firmware to a STM32
 * `:dfu-util` - The default command for flashing to STM32 devices. 
 * `:dfu-util-wait` - This works like the default command, but it gives you a (configurable) 10 second timeout before it attempts to flash the firmware.  You can use `TIME_DELAY=20` from the command line to change the timeout.
    * Eg: `make <keyboard>:<keymap>:dfu-util TIME_DELAY=5`
+* `:dfu-utilsplit-left` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Left Side" EEPROM setting for split keyboards.
```
Comment (fauxpark) [inline docs/flashing.md]:
```suggestion
* `:dfu-util-split-left` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Left Side" EEPROM setting for split keyboards.
```

File: docs/flashing.md [hunk 5]
```diff
@@ -174,4 +174,6 @@ There are a number of DFU commands that you can use to flash firmware to a STM32
...
+* `:dfu-utilsplit-right` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Right Side" EEPROM setting for split keyboards.
```
Comment (fauxpark) [inline docs/flashing.md]:
```suggestion
* `:dfu-util-split-right` - This flashes the normal firmware, just like the default option (`:dfu-util`). However, this also configures the "Right Side" EEPROM setting for split keyboards.
```

File: tmk_core/common/eeconfig.c [hunk 6]
```diff
@@ -48,6 +48,13 @@ void eeconfig_init_quantum(void) {
     eeprom_update_dword(EECONFIG_RGB_MATRIX, 0);
     eeprom_update_byte(EECONFIG_RGB_MATRIX_SPEED, 0);
 
+    // TODO: Remove once ARM has a way to configure EECONFIG_HANDEDNESS
+    //        within the emulated eeprom via dfu-util or another tool
+#ifdef INIT_EE_HANDS_LEFT
```
Comment (drashna) [inline tmk_core/common/eeconfig.c]: What happens if this was set incorrectly, and then reflashed? 

Should probably set it to 0, for ARM split boards.. 

eg: 
```c
#ifdef __arm__
#    ifdef INIT_EE_HANDS_LEFT
#    pragma message "Faking EE_HANDS for left hand"
    eeprom_update_byte(EECONFIG_HANDEDNESS, 1);
#    else
#    pragma message "Faking EE_HANDS for right hand"
    eeprom_update_byte(EECONFIG_HANDEDNESS, 0);
#    endif
#endif
```

Comment (zvecr) [reply tmk_core/common/eeconfig.c]: Tried to not lock down the logic too much so it *could* be used for non dfu-util non arm if required. For constancy I agree its best to init the right hand till eeprom memory is zeroed on all platforms.

**Comment by drashna** [comment]
Looks good!

**Pull request merged**